### ISSUE TITLE
Warnings with static build and go 1.9

### ISSUE THREAD

**Issue by yongtang** [opencontainers/runc#1576]
When building static (`make static`) with go 1.9, the following warnings show up:
```
[ec2-user@ip-172-31-9-144 runc]$ docker run --rm -i -t -v $PWD:/go/src/github.com/opencontainers/runc -w /go/src/github.com/opencontainers/runc golang:1.9
root@f4b077232050:/go/src/github.com/opencontainers/runc# apt-get -qq update && apt-get -qq install libseccomp-dev
...
...
root@f4b077232050:/go/src/github.com/opencontainers/runc# make static
CGO_ENABLED=1 go build  -tags "seccomp cgo static_build" -ldflags "-w -extldflags -static -X main.gitCommit="1c81e2a794c6e26a4c650142ae8893c47f619764" -X main.version=1.0.0-rc4+dev " -o runc .
# github.com/opencontainers/runc
/tmp/go-link-113476657/000007.o: In function `_cgo_a5acef59ed3f_Cfunc_dlopen':
/tmp/go-build/github.com/opencontainers/runc/vendor/github.com/coreos/pkg/dlopen/_obj/cgo-gcc-prolog:76: warning: Using 'dlopen' in statically linked applications requires at runtime the shared libraries from the glibc version used for linking
/tmp/go-link-113476657/000001.o: In function `_cgo_b0c710f30cfd_C2func_getaddrinfo':
/tmp/go-build/net/_obj/cgo-gcc-prolog:46: warning: Using 'getaddrinfo' in statically linked applications requires at runtime the shared libraries from the glibc version used for linking
CGO_ENABLED=1 go build  -tags "seccomp cgo static_build" -ldflags "-w -extldflags -static -X main.gitCommit="1c81e2a794c6e26a4c650142ae8893c47f619764" -X main.version=1.0.0-rc4+dev " -o contrib/cmd/recvtty/recvtty ./contrib/cmd/recvtty
# github.com/opencontainers/runc/contrib/cmd/recvtty
/tmp/go-link-355596637/000005.o: In function `_cgo_a5acef59ed3f_Cfunc_dlopen':
/tmp/go-build/github.com/opencontainers/runc/vendor/github.com/coreos/pkg/dlopen/_obj/cgo-gcc-prolog:76: warning: Using 'dlopen' in statically linked applications requires at runtime the shared libraries from the glibc version used for linking
/tmp/go-link-355596637/000000.o: In function `_cgo_b0c710f30cfd_C2func_getaddrinfo':
/tmp/go-build/net/_obj/cgo-gcc-prolog:46: warning: Using 'getaddrinfo' in statically linked applications requires at runtime the shared libraries from the glibc version used for linking
root@f4b077232050:/go/src/github.com/opencontainers/runc# 
```

The `'getaddrinfo'` could be addressed with using `netgo` tag.

**Comment by yongtang** [comment]
Created PR #1577 for `netgo` / `getaddrinfo`

**Comment by yongtang** [comment]
The `dlopen` warning points to `coreos/go-systemd` which uses tag `cgo` to switch on/off. The tag `cgo` may have an impact to `seccomp`, `sysconfig`, `nsenter` so further investigation is needed.

**Comment by yongtang** [comment]
Created PR #1579 for `dlopen` warning.

**Comment by thaJeztah** [comment]
both PRs are merged now

**Comment by yongtang** [comment]
Thanks @thaJeztah! Close the issue as it has been fixed by #1579 and #1577.
